File: src/main/java/cal/fstg/java/service/EmployerService.java
//package cal.fstg.java.service;
//
//import cal.fstg.java.dao.EmployerRepository;
//import cal.fstg.java.dao.ResumeRepository;
//import cal.fstg.java.model.Employer;
//import cal.fstg.java.model.Resume;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.stereotype.Service;
//
//import java.util.List;
//import java.util.Optional;
//
//@Service
//public class EmployerService {
//
//    private final EmployerRepository employerRepository;
//
//    @Autowired
//    public EmployerService(EmployerRepository employerRepository) {
//        this.employerRepository = employerRepository;
//    }
//
//    public List<Employer> getAll() {
//        return employerRepository.findAll();
//    }
//
//    public Optional<Employer> getOne(long id) {
//        return employerRepository.findById(id);
//    }
//
//    public Employer save(Employer employer) {
//        return employerRepository.save(employer);
//    }
//
//    public void delete(long id) {
//        employerRepository.deleteById(id);
//    }
//}

File: src/main/kotlin/cal/fstg/stagerkotlin/service/EmployerService.kt
package cal.fstg.stagerkotlin.service

import cal.fstg.stagerkotlin.dao.EmployerRepository
import cal.fstg.stagerkotlin.model.Employer
import cal.fstg.stagerkotlin.model.User
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.reactor.flux
import kotlinx.coroutines.reactor.mono
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class EmployerService(@Autowired private val employerRepository: EmployerRepository) {

    @ExperimentalCoroutinesApi
    fun getAll() = flux<Employer> { employerRepository.findAll() }

    fun getOne(id: String) = mono { employerRepository.findById(id) }

    fun save(employer: Employer) = mono { employerRepository.save(employer) }

    fun delete(id: String) = employerRepository.deleteById(id)
}